### PR TITLE
Change the failed duplicate only if the filename is the same

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
Ready

## Related Issues
fixes: link to the issue

## Description
Since there are some duplicate PB with the same ID as others ( deprecate and non-deprecate PB)
for example
[Deprecate PB](https://github.com/demisto/content/blob/master/Packs/BruteForce/Playbooks/Brute_Force_Investigation_-_Generic.yml)
[non-deprecated PB](https://github.com/demisto/content/blob/master/Packs/BruteForce/Playbooks/Brute_Force_Investigation_-_Generic_6_5.yml
)
We shouldn't fail the build on duplicate in such a case

## Screenshots
Paste here any images that will help the reviewer

### CODE REVIEW

**Review by tkatzir — APPROVED** [review]
Good catch!

File: content-repo/gendocs.py [hunk 1]
```diff
@@ -136,6 +139,34 @@ class DeprecatedInfo(TypedDict, total=False):
...
+            to_check (DocInfo): The DocInfo object to check.
+            check_with (DocInfo): The DocInfo object to check with.
+        Returns:
+             True if there is a mismatch in the versions otherwise False.
+    """
+    if to_check.id != check_with.id:
+        return False
+    if to_check.to_version < check_with.from_version or check_with.to_version < to_check.from_version:
+        return False
+    print(f'Found version conflict in {to_check.readme} and {check_with.readme} with the same ID {to_check.id}')
```
Comment (kobymeir) [inline content-repo/gendocs.py:166]: perhaps , add to the logs the versions?

File: content-repo/gendocs.py [hunk 2]
```diff
@@ -136,6 +139,34 @@ class DeprecatedInfo(TypedDict, total=False):
...
+def version_conflict(to_check: DocInfo, check_with: DocInfo):
+    """
+        Retrieves two DocInfo classes with the same ID and returns if there is a conflict between the versions.
+        Args:
+            to_check (DocInfo): The DocInfo object to check.
+            check_with (DocInfo): The DocInfo object to check with.
+        Returns:
+             True if there is a mismatch in the versions otherwise False.
+    """
+    if to_check.id != check_with.id:
```
Comment (kobymeir) [inline content-repo/gendocs.py:162]: please join the if statements with a `or`

File: content-repo/gendocs.py [hunk 3]
```diff
@@ -434,6 +467,8 @@ def process_extra_readme_doc(target_dir: str, prefix: str, readme_file: str, pri
         name = yml_data['title']
         file_id = yml_data.get('id') or normalize_id(name)
         desc = yml_data.get('description')
+        from_version = yml_data.get('fromversion')
```
Comment (kobymeir) [inline content-repo/gendocs.py:470]: are we sure that all yml files are valid and containing the from/to version definition?

Comment (MosheEichler) [reply content-repo/gendocs.py:470]: From the [schemas](https://github.com/demisto/demisto-sdk/tree/master/demisto_sdk/commands/common/schemas), it looks like only playbooks and only the fromversion is required while toversion or integrations and scripts no

I'll add by default the from version to 0 and to version to 9999 is this good enough?

File: content-repo/gendocs.py [hunk 4]
```diff
@@ -136,6 +139,35 @@ class DeprecatedInfo(TypedDict, total=False):
     note: str
 
 
+def version_convertor(version: str) -> int:
```
Comment (kobymeir) [inline content-repo/gendocs.py:142]: i'm against parsing the version to int. as 6.10.10 < 7.0.0. but you conversion to number will be 61010 and 700 which will cause and error. please use SemVer comparison using this https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python

**Review by kobymeir — CHANGES_REQUESTED** [review]

**Review by kobymeir — APPROVED** [review]

**Pull request merged**